### DISCUSSION TITLE
Re-evaluate middlewares and/or soft-reload current page

### DISCUSSION THREAD

**Discussion by some-user123** [nuxt/nuxt#20842] [Questions]
I'm rending pages with different content depending on the authentication state. Some pages will forward to the sign in page, if there is no authentication, others don't care about authentications, some will show different content depending on the authentication state.

When user is signing out, I'd like to re-evaluate all middlewares for the current page and/or soft-reload the current page. As the logout happens in a parent component, I can't just call `refresh` of `useFetch` (or something similar).
Of course, I could hard-reload the current page. But that doesn't seem like a nice solution. Is there a better way to programmatically trigger a re-evaluation of the current page's middlewares?

**Comment by warflash** [comment]
Force pushing the route you're on does trigger middleware, so `router.push('/my-page', { force: true })` should do the job

**Reply by some-user123** [reply]
Using typescript, there is no second parameter allowed.
`router.push({ path: '...', force: true })` is possible though. Thanks!

**Reply by warflash** [reply]
Oh yeah, slightly misremembered that it seems :) You're welcome, glad it worked out!
